First draft of the mentorship program page (#1043)
nf-core/nf-co.re: markdown/mentorships.md

## Introduction

In 2022, Nextflow and nf-core are starting a joint program of mentorships.
We hope to organise a small number of mentoring pairs to improve outreach and training in underrepresented groups and areas.

> Mentors will recieve a small stipend in return for helping their mentee.

Reviewer: ```suggestion
Mentors will receive a small stipend in return for helping their mentee.
```

- Mentors will be paid with a voucher equivalent to $800 USD upon completion of the mentorship round (subject to completion of certain milestones)
- A commitment of at least 2 hours a week is expected from both mentor and mentee
- Mentors may participate in multiple rounds of the mentorship program,
- Mentees may only join for one rond (though are welcome back as a mentor!)

There will be a set of milestones and aims for each partnership and a short written report upon completion.

### Scope

The exact scope and aims of each mentoring partnership can vary according to your needs and situation.
These will be judged on a case-by-case basis as part of the application process.

Some typical aims could be:

- Setting up Nextflow and nf-core in your institution
  - Getting requirements up and running on the local compute infrastructure
  - Developing an institutional config for you and others to use
  - Testing and scaling up the use of nf-core pipelines
- Learning how to make a contribution to nf-core
  - Developing new and / or contributing to existing pipeline(s)
  - Working with nf-core/tools or the website

>   - Presenting a nf-core/bytesize talk about a chosen topic

Reviewer: ```suggestion
  - Presenting a nf-core/bytesize talk about a chosen topic
  - Writing an extensive tutorial about under-described aspects of nf-core 
```

### Prerequisites

- Mentees

>   - Required: Some familiarity with the linux command line

Reviewer 2: ```suggestion
  - Required: familiarity with the linux command line, git and github
```

  - Suggested: Basic bioinformatics experience
- Mentors

>   - Required: Good experience with running Nextflow pipelines

Reviewer 2: ```suggestion
  - Required: Good experience with running and developing Nextflow pipelines
  - Required: Experience with container engines (docker, singularity)
```

>   - Suggested: Experience with developing Nextflow pipelines

Reviewer 2: ```suggestion
```

These will be judged on a case-by-case basis as part of the application process.

### Organisation

In summary (all points subject to change):

- We will start with rounds of 5 mentor + mentee pairs
- Mentorships will run for 4 months
- New rounds approximately every 6 months
- Mentees will get free tutoring from an experienced developer about using Nextflow and nf-core
- Mentors will be paid with a voucher equivalent to $800 USD upon completion of the mentorship round (subject to completion of certain milestones)
- A commitment of at least 2 hours a week is expected from both mentor and mentee
- Mentors may participate in multiple rounds of the mentorship program,
- Mentees may only join for one rond (though are welcome back as a mentor!)

There will be a set of milestones and aims for each partnership and a short written report upon completion.

### Prerequisites

- Mentees
  - Required: Familiarity with the linux command line, Git and GitHub
  - Suggested: Basic bioinformatics experience
- Mentors
  - Required: Good experience with running and developing Nextflow pipelines
  - Required: Experience with container engines (docker, singularity, ...)

>   - Suggested: Experience of contributing to the nf-core project

Author: ```suggestion
  - Suggested: Previous contributions to the nf-core project
```
